src/web/mjs/engine/DiscordPresence.mjs

```diff
@@ -101,14 +101,16 @@ export default class DiscordPresence {
         }
     }
 
-    stopDiscordPresence() {
+    stopDiscordPresence(reset = false) {
+        // Setting reset to true will assume a proper connection was already lost
+
         this.statusNew = false;
         clearInterval(this.updater);
-        if (this.rpc) {
+        if (this.rpc && !reset) {
             this.rpc.clearActivity();
             this.rpc.destroy();
-            this.rpc = null;
         }
+        this.rpc = null;
     }
 
     async startDiscordPresence() {
```

Comment:
Method complexity is growing, consider outsourcing conditional blocks to separate methods

Reply:
will keep it mind for the future. for now: won't change